lib/index.js

```diff
@@ -89,44 +88,81 @@ AdWords.prototype.page = function(page) {
  */
 
 AdWords.prototype.track = function(track) {
-  var id = this.options.conversionId;
+  var self = this;
   var props = track.properties();
-  var remarketing = this.options.remarketing;
-  var events = this.events(track.event());
+  var eventMappings = this.options.eventMappings;
   var revenue = track.revenue() || 0;
-  // Check if this is a whitelisted event for standalone remarketing tag
-  var whitelisted = this.options.whitelist.indexOf(track.event()) > -1;
-  var sentAlready = false;
 
-  each(function(label) {
-    delete props.revenue;
-    // Fire conversion tag
-    window.google_trackConversion({
-      google_conversion_id: id,
-      google_custom_params: props,
-      google_conversion_language: 'en',
-      google_conversion_format: '3',
-      google_conversion_color: 'ffffff',
-      google_conversion_label: label,
-      google_conversion_value: revenue,
-      google_remarketing_only: false // ensure this is a conversion tag
+  if (eventMappings) { // if eventMappings exists then the new metadata schema is being used
+    eventMappings.forEach(function(mapping) {
+      if (mapping.value.eventName.toLowerCase() !== track.event().toLowerCase()) return;
+      var id = mapping.value.conversionId === '' ?  self.options.conversionId : mapping.value.conversionId;  // customer can either specify one global conversion id or one per event<->label mapping
```

Comment:
Unsure what types there are here, but isn't this equivalent to `mapping.value.conversionId || self.options.conversionId`?

Reply:
mapping.value.conversionId is a `string` type metadata settings so if it's left blank (which is. what I'm checking for here) I don't want to use it

Reply:
Right, but '' resolves to `false` in comparisons, no?

Reply:
just checked you're right. fixing now